gtpv1/constants.go

```diff
@@ -80,6 +82,129 @@ const (
+	case ResCauseMandatoryIEMissing:
+		causeStr = "MandatoryIEMissing"
+	case ResCauseOptionalIEIncorrect:
+		causeStr = "OptionalIEIncorrect"
+	case ResCauseSystemFailure:
+		causeStr = "SystemFailure"
+	case ResCauseRoamingRestriction:
+		causeStr = "RoamingRestriction"
+	case ResCausePTMSISignatureMismatch:
+		causeStr = "PTMSISignatureMismatch"
+	case ResCauseGPRSConnectionSuspended:
+		causeStr = "GPRSConnectionSuspended"
+	case ResCauseAuthenticationFailure:
+		causeStr = "AuthenticationFailure"
+	case ResCauseUserAuthenticationFailed:
+		causeStr = "UserAuthenticationFailed"
+	case ResCauseContextNotFound:
+		causeStr = "ContextNotFound"
+	case ResCauseAllDynamicPDPAddressesAreOccupied:
+		causeStr = "AllDynamicPDPAddressesAreOccupied"
+	case ResCauseNoMemoryIsAvailable:
+		causeStr = "NoMemoryIsAvailable"
+	case ResCauseRelocationFailure:
+		causeStr = "RelocationFailure"
+	case ResCauseUnknownMandatoryExtensionHeader:
+		causeStr = "UnknownMandatoryExtensionHeader"
+	case ResCauseSemanticErrorInTheTFTOperation:
+		causeStr = "SemanticErrorInTheTFTOperation"
+	case ResCauseSyntacticErrorInTheTFTOperation:
+		causeStr = "SyntacticErrorInTheTFTOperation"
+	case ResCauseSemanticErrorsInPacketFilter:
+		causeStr = "SemanticErrorsInPacketFilter"
+	case ResCauseSyntacticErrorsInPacketFilter:
+		causeStr = "SyntacticErrorsInPacketFilter"
+	case ResCauseMissingOrUnknownAPN:
+		causeStr = "MissingOrUnknownAPN"
+	case ResCauseUnknownPDPAddressOrPDPType:
+		causeStr = "UnknownPDPAddressOrPDPType"
+	case ResCausePDPContextWithoutTFTAlreadyActivated:
+		causeStr = "PDPContextWithoutTFTAlreadyActivated"
+	case ResCauseAPNAccessDeniedNoSubscription:
+		causeStr = "APNAccessDeniedNoSubscription"
+	case ResCauseAPNRestrictionTypeIncompatibilityWithCurrentlyActivePDPContexts:
+		causeStr = "APNRestrictionTypeIncompatibilityWithCurrentlyActivePDPContexts"
+	case ResCauseMSMBMSCapabilitiesInsufficient:
+		causeStr = "MSMBMSCapabilitiesInsufficient"
+	case ResCauseInvalidCorrelationID:
+		causeStr = "InvalidCorrelationID"
+	case ResCauseMBMSBearerContextSuperseded:
+		causeStr = "MBMSBearerContextSuperseded"
+	case ResCauseBearerControlModeViolation:
+		causeStr = "BearerControlModeViolation"
+	case ResCauseCollisionWithNetworkInitiatedRequest:
+		causeStr = "CollisionWithNetworkInitiatedRequest"
+	case ResCauseAPNCongestion:
+		causeStr = "APNCongestion"
+	case ResCauseBearerHandlingNotSupported:
+		causeStr = "BearerHandlingNotSupported"
+	case ResCauseTargetAccessRestrictedForTheSubscriber:
+		causeStr = "TargetAccessRestrictedForTheSubscriber"
+	case ResCauseUEIsTemporarilyNotReachableDueToPowerSaving:
+		causeStr = "UEIsTemporarilyNotReachableDueToPowerSaving"
+	case ResCauseRelocationFailureDueToNASMessageRedirection:
+		causeStr = "RelocationFailureDueToNASMessageRedirection"
+	// Skipped range 234-255: for future use / reserved for prime.
+	default:
+		causeStr = strconv.FormatUint(uint64(cause), 10)
+	}
+	return causeStr
+}
```

Comment:
Would it be better to use map?

Reply:
or is there c-style conversion in GO like: `##constname` would convert to "constname" static string

Reply:
Agree with @dzvancuks that using map is better.

Reply:
Why?

Reply:
I think it is more flexible to use.